app/src/ui/repository-settings/git-ignore.tsx

```diff
@@ -24,6 +93,8 @@ export class GitIgnore extends React.Component<IGitIgnoreProps, {}> {
           </LinkButton>
         </p>
 
+        {this.state.gitIgnoreEmpty && this.renderTemplates()}
```

Comment:
@thenewpotato it's a bit nitpicky 😄 but usually, when there is a conditional rendering of an element, we prefer the condition to be part of the element itself. 

So you can change line 96 to `{this.renderTemplates()}` and add a statement in the function's body that checks `this.state.gitIgnoreEmpty`

You can use `create-repository.tsx` again as an example of what I'm referring to. If the conditions to show `renderInvalidPathError` and `renderGitRepositoryWarning` are not met the functions return `null`